modules/programs/zsh/default.nix

```diff
@@ -412,15 +412,15 @@ in
 
         (mkIf (dotDirAbs != homeDir) {
           home.file."${dotDirRel}/.zshenv".text = ''
-            export ZDOTDIR=${dotDirAbs}
+            export ZDOTDIR=${lib.escapeShellArg dotDirAbs}
```

Comment:
There's definitely an `exportVar` (unsure about the spelling) function under `lib.hm`